lib/fotofetch.rb

```diff
@@ -59,25 +81,27 @@ def height_ok?(sizes)
       (0 < height && height < sizes[1][1]) || height > (sizes[1][1] * 2)
     end
 
+    def root_url(link)
+      link.split("/")[2]
+    end
+
     # Adds root urls as hash keys
     def add_sources(urls)
-      results = {}.compare_by_identity
-      urls.each { |link| results[link.split("/")[2]] = link}
-      results
+      urls.map { |link| [root_url(link), link] }.to_h
```

Comment:
@plato721 when I run the tests, we go from 5 links to 4 links in this map.to_h. One of the keys must repeat. So that one problem aside, all the refactors look good, I'll merge this in if you can get that test passing (test_it_finds_image_links). Thanks.